onnxruntime/test/shared_lib/test_inference.cc

```diff
@@ -87,14 +87,24 @@ static void TestInference(Ort::Env& env, const std::basic_string<ORTCHAR_T>& mod
                           OrtCustomOpDomain* custom_op_domain_ptr,
                           const char* custom_op_library_filename,
                           void** library_handle = nullptr,
-                          bool test_session_creation_only = false) {
+                          bool test_session_creation_only = false,
+                          void* cuda_compute_stream = nullptr) {
   Ort::SessionOptions session_options;
 
   if (provider_type == 1) {
 #ifdef USE_CUDA
-    Ort::ThrowOnError(OrtSessionOptionsAppendExecutionProvider_CUDA(session_options, 0));
     std::cout << "Running simple inference with cuda provider" << std::endl;
+    OrtCUDAProviderOptions cuda_options{
+        0,
+        OrtCudnnConvAlgoSearch::EXHAUSTIVE,
+        std::numeric_limits<size_t>::max(),
+        0,
+        true,
+        cuda_compute_stream != nullptr ? 1 : 0,
+        cuda_compute_stream != nullptr ? cuda_compute_stream : nullptr};
+    session_options.AppendExecutionProvider_CUDA(cuda_options);
 #else
```

Comment:
nit: May be useful to have a helper that creates the default options with optional compute stream.

Reply:
just created a helper method for now